client/app/pages/dashboards/components/DashboardHeader.jsx

```diff
@@ -0,0 +1,272 @@
+      autoFocusButton: null,
+    });
+  };
+
+  return (
+    <Dropdown
+      trigger={["click"]}
+      placement="bottomRight"
+      overlay={
+        <Menu data-test="DashboardMoreButtonMenu">
+          <Menu.Item className={cx({ hidden: gridDisabled })}>
+            <a onClick={() => setEditingLayout(true)}>Edit</a>
+          </Menu.Item>
+          {clientConfig.showPermissionsControl && isDashboardOwnerOrAdmin && (
+            <Menu.Item>
+              <a onClick={managePermissions}>Manage Permissions</a>
+            </Menu.Item>
+          )}
+          {!dashboard.is_draft && (
+            <Menu.Item>
+              <a onClick={togglePublished}>Unpublish</a>
+            </Menu.Item>
+          )}
+          <Menu.Item>
+            <a onClick={archive}>Archive</a>
+          </Menu.Item>
+        </Menu>
+      }>
+      <Button className="icon-button m-l-5" data-test="DashboardMoreButton">
+        <Icon type="ellipsis" rotate={90} />
+      </Button>
+    </Dropdown>
+  );
+}
+
+DashboardMoreOptionsButton.propTypes = {
+  dashboardOptions: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+};
+
+function DashboardControl({ dashboardOptions }) {
+  const {
+    dashboard,
+    togglePublished,
+    canEditDashboard,
+    fullscreen,
+    toggleFullscreen,
+    showShareDashboardDialog,
+  } = dashboardOptions;
+  const showPublishButton = dashboard.is_draft;
+  const showRefreshButton = true;
+  const showFullscreenButton = !dashboard.is_draft;
+  const showShareButton = dashboard.publicAccessEnabled || (canEditDashboard && !dashboard.is_draft);
+  const showMoreOptionsButton = canEditDashboard;
+  return (
+    <div className="dashboard-control">
+      {!dashboard.is_archived && (
+        <span className="hidden-print">
+          {showPublishButton && (
+            <Button className="m-r-5 hidden-xs" onClick={togglePublished}>
+              <span className="fa fa-paper-plane m-r-5" /> Publish
+            </Button>
+          )}
+          {showRefreshButton && <RefreshButton dashboardOptions={dashboardOptions} />}
+          {showFullscreenButton && (
+            <Tooltip className="hidden-xs" title="Enable/Disable Fullscreen display">
+              <Button type={buttonType(fullscreen)} className="icon-button m-l-5" onClick={toggleFullscreen}>
+                <i className="zmdi zmdi-fullscreen" />
+              </Button>
+            </Tooltip>
+          )}
+          {showShareButton && (
+            <Tooltip title="Dashboard Sharing Options">
+              <Button
+                className="icon-button m-l-5"
+                type={buttonType(dashboard.publicAccessEnabled)}
+                onClick={showShareDashboardDialog}
+                data-test="OpenShareForm">
+                <i className="zmdi zmdi-share" />
+              </Button>
+            </Tooltip>
+          )}
+          {showMoreOptionsButton && <DashboardMoreOptionsButton dashboardOptions={dashboardOptions} />}
```

Comment:
Now that the controls are wrapped for mobiles, I'm not hiding those buttons anymore (I saw no reason to keep doing it)

Reply:
Let's continue hiding the full screen button as it's meaningless on mobile.

Reply:
That one is still hidden, only the Share and the Dropdown were changes.